Reported upstream: https://github.com/owncloud/client/pull/8684

From 89e59a352063f372a1573d326a80bd79ff4d5c37 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@gmail.com>
Date: Sat, 5 Jun 2021 13:27:13 -0300
Subject: [PATCH] Fix use of deleted function

Compilation fails due to implicit cast to QChar

Compilation on void linux (qt 5.15.3+20210429) fails with
csync_exclude.cpp:489:34: error: use of deleted function 'QCharRef& QCharRef::operator=(char)'
---
 src/csync/csync_exclude.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/csync/csync_exclude.cpp b/src/csync/csync_exclude.cpp
index 586035346b7..c5c7bbccf5c 100644
--- a/src/csync/csync_exclude.cpp
+++ b/src/csync/csync_exclude.cpp
@@ -488,7 +488,7 @@ QString ExcludedFiles::convertToRegexpSyntax(QString exclude, bool wildcardsMatc
             // Translate [! to [^
             QString bracketExpr = exclude.mid(i, j - i + 1);
             if (bracketExpr.startsWith(QLatin1String("[!")))
-                bracketExpr[1] = '^';
+                bracketExpr[1] = QLatin1Char('^');
             regex.append(bracketExpr);
             i = j;
             break;
